Makefile

```diff
@@ -165,146 +165,81 @@ protoc:
 
-	mkdir -p mocks/lib/torrent/scheduler
-	$(mockgen) \
-		-destination=mocks/lib/torrent/scheduler/mockscheduler.go \
-		-package mockscheduler \
-		github.com/uber/kraken/lib/torrent/scheduler ReloadableScheduler,Scheduler
+	$(call kraken_mockgen,lib/store,mockstore,FileReadWriter)
 
-	mkdir -p mocks/origin/blobclient
-	$(mockgen) \
-		-destination=mocks/origin/blobclient/mockblobclient.go \
-		-package mockblobclient \
-		github.com/uber/kraken/origin/blobclient \
-		Client,Provider,ClusterClient,ClusterProvider,ClientResolver
+	$(call kraken_mockgen,lib/torrent/scheduler,mockscheduler,ReloadableScheduler)
+	$(call kraken_mockgen,lib/torrent/scheduler,mockscheduler,Scheduler)
 
-	mkdir -p mocks/lib/dockerregistry/transfer
-	$(mockgen) \
-		-destination=mocks/lib/dockerregistry/transfer/mocktransferer.go \
-		-package mocktransferer \
-		github.com/uber/kraken/lib/dockerregistry/transfer ImageTransferer
+	$(call kraken_mockgen,origin/blobclient,mockblobclient,Client)
+	$(call kraken_mockgen,origin/blobclient,mockblobclient,Provider)
+	$(call kraken_mockgen,origin/blobclient,mockblobclient,ClusterClient)
+	$(call kraken_mockgen,origin/blobclient,mockblobclient,ClusterProvider)
+	$(call kraken_mockgen,origin/blobclient,mockblobclient,ClientResolver)
 
-	mkdir -p mocks/tracker/metainfoclient
-	$(mockgen) \
-		-destination=mocks/tracker/metainfoclient/mockmetainfoclient.go \
-		-package mockmetainfoclient \
-		github.com/uber/kraken/tracker/metainfoclient Client
+	$(call kraken_mockgen,lib/dockerregistry/transfer,mocktransferer,ImageTransferer)
 
-	mkdir -p mocks/os
-	$(mockgen) \
-		-destination=mocks/os/mockos.go \
-		-package mockos \
-		os FileInfo
+	$(call kraken_mockgen,tracker/metainfoclient,mockmetainfoclient,Client)
 
-	mkdir -p mocks/net/http
-	$(mockgen) \
-		-destination=mocks/net/http/mockhttp.go \
-		-package mockhttp \
-		net/http RoundTripper
+	$(call kraken_mockgen,lib/persistedretry,mockpersistedretry,Store)
+	$(call kraken_mockgen,lib/persistedretry,mockpersistedretry,Task)
+	$(call kraken_mockgen,lib/persistedretry,mockpersistedretry,Executor)
+	$(call kraken_mockgen,lib/persistedretry,mockpersistedretry,Manager)
 
-	mkdir -p mocks/lib/persistedretry
-	$(mockgen) \
-		-destination=mocks/lib/persistedretry/mockpersistedretry.go \
-		-package mockpersistedretry \
-		github.com/uber/kraken/lib/persistedretry Store,Task,Executor,Manager
+	$(call kraken_mockgen,lib/persistedretry/tagreplication,mocktagreplication,RemoteValidator)
 
-	mkdir -p mocks/lib/persistedretry/tagreplication
-	$(mockgen) \
-		-destination=mocks/lib/persistedretry/tagreplication/mocktagreplication.go \
-		-package mocktagreplication \
-		github.com/uber/kraken/lib/persistedretry/tagreplication RemoteValidator
+	mkdir -p mocks/os
+	$(mockgen) -destination=mocks/os/mockos.go -package mockos os FileInfo
+
+	mkdir -p mocks/net/http
+	$(mockgen) -destination=mocks/net/http/mockhttp.go -package mockhttp net/http RoundTripper
```

Comment:
To avoid this snowflake, we could alias this type in `httputil` and use the standard template to create a `mockhttputil` package instead:

```
package httputil

...

// Alias for mocking purposes.
type RoundTripper = http.RoundTripper
```

Reply:
Done.